psydac/linalg/block.py

```diff
@@ -1325,7 +1326,7 @@ def func(blocks, v, out, **args):
 #===============================================================================
 class BlockDiagonalSolver( LinearSolver ):
     """
-    A LinearSolver that can be written as blocks of other LinearSolvers,
+    A LinearSolver that can be written as blocks of other (Kronecker-)LinearSolvers,
```

Comment:
Is this class still used in the code?

Reply:
Yes, but only once in `gobal_projectors.py`:
```
        # build a BlockDiagonalSolver, if necessary
        if len(solverblocks) == 1:
            self._solver = solverblocks[0]
        else:
            self._solver = BlockDiagonalSolver(self._space.vector_space, blocks=solverblocks)
```

We should be able to replace the last line by
```
blocks = [[solverblocks[j] if j==i else None for j in range(len(solvers))] for i in range(len(solvers))]
self._solver = BlockLinearOperator(self._space.vector_space, self._space.vector_space, blocks)
```
If for example `len(solverblocks) == 3`, `blocks` looks like this:
```
blocks = [[solverblocks[0], None, None],
          [None, solverblocks[1], None],
          [None, None, solverblocks[2]]
```